salt/states/boto3_route53.py

```diff
@@ -603,6 +620,15 @@ def rr_present(name, HostedZoneId=None, DomainName=None, PrivateZone=False, Name
                 fixed_rrs += [rr]
         ResourceRecords = [{'Value': rr} for rr in sorted(fixed_rrs)]
 
+    # If HostedZoneId is missing from AliasTarget, provide it on our own.
+    if AliasTarget:
+        if not 'HostedZoneId' in AliasTarget:
+            cleanid = HostedZoneId[12:] if HostedZoneId.startswith('/hostedzone/') else HostedZoneId
+            AliasTarget.update({'HostedZoneId': cleanid})
+
+    if HealthCheckName is not None:
```

Comment:
Would this read better as just `if HealthCheckName:`? That will skip on `None`, but it will also skip `''` and any other false-y values.